### PR TITLE
[IND-391] Add geo-blocking logic.

### PR DESCRIPTION
**Pull request by vincentwschau**
If indexer level geo-blocking is enabled, all requests have to have Cloudflare country headers, or will return a 403 or have the websocket connection terminated.

### CODE REVIEW

**Review by dydxwill — COMMENTED** [review]
is it possible to merge the restrict-countries logic from `socks` and `comlink` into a shared module?

File: indexer/services/comlink/__tests__/lib/restrict-countries.test.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+describe('rejectRestrictedCountries', () => {
+  let isRestrictedCountrySpy: jest.SpyInstance;
+  let req: any;
+  let res: any;
+  let next: any;
+
+  const defaultEnabled: boolean = config.INDEXER_LEVEL_GEOBLOCKING_ENABLED;
+
+  beforeEach(() => {
+    config.INDEXER_LEVEL_GEOBLOCKING_ENABLED = true;
```
Comment (Christopher-Li) [inline indexer/services/comlink/__tests__/lib/restrict-countries.test.ts:22]: nit: put this and the other in beforeAll and afterAll?

**Comment by vincentwschau** [comment]
>is it possible to merge the restrict-countries logic from socks and comlink into a shared module?
It's possible, but I didn't want to create another package for it. There is the `compliance` module I can throw this into.

**Comment by dydxwill** [comment]
> > is it possible to merge the restrict-countries logic from socks and comlink into a shared module?
> > It's possible, but I didn't want to create another package for it. There is the `compliance` module I can throw this into.

up to you whether you want to do this, I think putting it in `compliance` module would be helpful

**Review by dydxwill — APPROVED** [review]

**Review by Christopher-Li — APPROVED** [review]

**Pull request merged**